src/gt4py/next/program_processors/runners/dace_fieldview/transformations/map_promoter.py

```diff
@@ -0,0 +1,379 @@
+        map_to_promote.range = subsets.Range(promoted_ranges)
+        map_to_promote.params = promoted_params
+
+    def missing_map_params(
+        self,
+        map_to_promote: nodes.Map,
+        source_map: nodes.Map,
+        be_strict: bool = True,
+    ) -> list[str] | None:
+        """Returns the parameter that are missing in the map that should be promoted.
+
+        The returned sequence is empty if they are already have the same parameters.
+        The function will return `None` is promoting is not possible.
+
+        Args:
+            map_to_promote: The map that should be promoted.
+            source_map: The map acting as template.
+            be_strict: Ensure that the ranges that are already there are correct.
+        """
+        source_params_set: set[str] = set(source_map.params)
+        curr_params_set: set[str] = set(map_to_promote.params)
+
+        # The promotion can only work if the source map's parameters
+        #  if a superset of the ones the map that should be promoted is.
+        if not source_params_set.issuperset(curr_params_set):
+            return None
+
+        if be_strict:
+            # Check if the parameters that are already in the map to promote have
+            #  the same range as in the source map.
+            source_ranges: subsets.Range = source_map.range
+            curr_ranges: subsets.Range = map_to_promote.range
+            curr_param_to_idx: dict[str, int] = {p: i for i, p in enumerate(map_to_promote.params)}
+            source_param_to_idx: dict[str, int] = {p: i for i, p in enumerate(source_map.params)}
+            for param_to_check in curr_params_set:
+                curr_range = curr_ranges[curr_param_to_idx[param_to_check]]
+                source_range = source_ranges[source_param_to_idx[param_to_check]]
+                if curr_range != source_range:
+                    return None
+        return list(source_params_set - curr_params_set)
+
+
+@properties.make_properties
+class SerialMapPromoter(BaseMapPromoter):
+    """Promote a map such that it can be fused serially.
+
+    A condition for fusing serial Maps is that they cover the same range. This
+    transformation is able to promote a Map, i.e. adding the missing dimensions,
+    such that the maps can be fused.
+    For more information see the `BaseMapPromoter` class.
+
+    Notes:
+        The transformation does not perform the fusing on its one.
+
+    Todo:
+        The map should do the fusing on its own directly.
+    """
+
+    # Pattern Matching
+    exit_first_map = transformation.transformation.PatternNode(nodes.MapExit)
+    access_node = transformation.transformation.PatternNode(nodes.AccessNode)
+    entry_second_map = transformation.transformation.PatternNode(nodes.MapEntry)
+
+    @classmethod
+    def expressions(cls) -> Any:
+        """Get the match expressions.
+
+        The function generates two different match expression. The first match
```

Comment:
different? they look exactly the same

Reply:
You are right 'different' is probably the wrong word.
You can see the reason in `map_to_promote()` and `source_map()` where the returned value depends on `self.expr_index` (which is defined by the pattern matching mechanism).
This is done to have only one code path.